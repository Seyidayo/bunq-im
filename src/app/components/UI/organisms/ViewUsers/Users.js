import { Fragment, useContext, useEffect, useState } from 'react';
import { NavLink } from 'react-router-dom';

import DataContext from 'app/context';
import { parseTime } from 'app/utils/helpers';
import Loader from 'app/components/UI/atoms/Loader/Loader';
import Navbar from 'app/components/UI/molecules/Navbar/Navbar';

const Users = () => {
  const { users, fetchUsers } = useContext(DataContext);
  const [status, setStatus] = useState('idle');

  // Fetch all users
  useEffect(() => {
    async function fetchData() {
      try {
        setStatus('loading');
        await fetchUsers();
        setStatus('fetched');
      } catch {
        setStatus('error');
      }
    }
    fetchData();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <aside className="app__sidebar">
      <Navbar />
      {status === 'loading' && !users.length ? (
        <Loader />
      ) : status === 'fetched' && users.length < 1 ? (
        <div className="empty__container">
          <h2>No one's here at the moment!</h2>
          <span>Check back later</span>
        </div>
      ) : (
        users.map((user) => (
          <Fragment key={user.id}>
            <NavLink
              to={`/im/${user.id}`}
              className="app__sidebar__link"
              activeClassName="active"
            >
              <img
                className="user__avatar"
                width="50px"
                height="50px"
                loading="lazy"
                src={`https://i.pravatar.cc/50?img=${user.id}`}
                alt={user.name}
              />
              <p className="user__name"> {user.name}</p>
              <span className="user__last-seen">
                {parseTime(user.last_seen_at)}
              </span>
            </NavLink>
          </Fragment>
        ))
      )}
    </aside>
  );
};

export default Users;
