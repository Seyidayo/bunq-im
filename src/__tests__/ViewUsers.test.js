import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import ViewUsers from 'app/components/UI/organisms/ViewUsers';

describe('<ViewUsers />', () => {
  it('It renders correctly and matches snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <ViewUsers />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
