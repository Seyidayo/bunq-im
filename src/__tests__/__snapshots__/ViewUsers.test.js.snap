// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewUsers /> It renders correctly and matches snapshot 1`] = `
<aside
  className="app__sidebar"
>
  <header
    className="header text-orange navbar__header"
  >
    <a
      href="/im"
      onClick={[Function]}
    >
      <h1
        className="h4"
      >
        bunqIM
      </h1>
    </a>
    <ul>
      <li />
      <li>
        <button
          className="secondary"
          onClick={[Function]}
          title="Logout"
          type="button"
        >
          LG
        </button>
      </li>
    </ul>
  </header>
</aside>
`;
